packages/grafana-ui/src/components/uPlot/Plot.tsx

```diff
@@ -83,12 +83,7 @@ export class UPlotChart extends Component<PlotProps, UPlotChartState> {
   componentDidUpdate(prevProps: PlotProps) {
     let { plot } = this.state;
 
-    if (!sameDims(prevProps, this.props)) {
-      plot?.setSize({
-        width: Math.floor(this.props.width),
-        height: Math.floor(this.props.height),
-      });
-    } else if (!sameConfig(prevProps, this.props)) {
+    if (!sameDims(prevProps, this.props) || !sameConfig(prevProps, this.props)) {
       this.reinitPlot();
```

Comment:
Does reinitPlot() not also calculate the size?

Reply:
More specifically, @leeoniya, is there a reason for optimization to separate these two conditions?

Reply:
yes. you dont want to reinit the plot on every pixel of a panel resize!

the solution should be to make sure that the stat panel re-inits the config when the min or max change.